src/config.rs

```diff
@@ -1456,10 +1498,10 @@ pub async fn reload_config(client_server_map: ClientServerMap) -> Result<bool, E
     };
 
     if old_config != new_config {
-        info!("Config changed, reloading");
-        ConnectionPool::from_config(client_server_map).await?;
```

Comment:
Why did you remove this from here and copy/pasted it in 3 other locations instead? It seems like you'd want to just reload the connection pools in every case now, which is fine, since we don't re-initialize connection pools if the configuration hasn't changed. Maybe we should just call `ConnectionPool::from_config(...)` here and not call it independently everywhere else in the code?

Reply:
> Why did you remove this from here and copy/pasted it in 3 other locations instead? It seems like you'd want to just reload the connection pools in every case now, which is fine, since we don't re-initialize connection pools if the configuration hasn't changed. Maybe we should just call `ConnectionPool::from_config(...)` here and not call it independently everywhere else in the code?

Since certificates can be renewed, isn't it required to reload the pools afterwards?

Reply:
@levkk, please reply me

Reply:
Apologies, been very busy this week. I'll take another look at this. I think this is pretty much ready to go, just need to give it one more look.

Reply:
> Apologies, been very busy this week. I'll take another look at this. I think this is pretty much ready to go, just need to give it one more look.

Did you watch my PR?

Reply:
My question is not why you need to reload pools but why did you remove the code that does it automatically and copy/pasted it in 3 different places that do the same thing as before?

Reply:
https://github.com/WagnerPMC/pgcat/pull/1